### ISSUE TITLE
G0 - ckeditor changes- to improve accessibility

### ISSUE THREAD

**Issue by anncesar** [ccswbs/hjckrrh#174]
The ckeditor continues to be a problem for creating accessible content, and while we may be able to train clients, we should consider doing the following changes to help ensure users create accessible content:

1) Remove defaults from Image/table upon inserting
2) also Consider removing the following options, if possible from the editor:
Image:  Wdith, Height, Border
Table: Border size, width, height, cell spacing, cell padding
3) consider removing the current insert table and insert image options and creating our own custom buttons that insert tables and images properly (using Bootstrap classes only).
4) we may also want to consider getting a license for this too: https://cksource.com/ckeditor/services#accessibility-checker ---
Possibility-limit the output on filter to just appropriate html tags and bootstrap classes.  Other research may be required, look to drupal 8 also?

**Comment by mmafe** [comment]
Original comment from @brharp : "@mmafe Please test. I imported the wysiwyg filter configuration from our old SVN/websolutions platform, and updated it to allow only bootstrap classes. I think we may need to expand the list of allowed tags. Looking for ideas."

**Comment by mmafe** [comment]
**Test Notes**

I'm seeing the following...

![wysiwyg-update-1 0](https://cloud.githubusercontent.com/assets/1927902/11575472/38526b40-99df-11e5-9d82-7899765f06b2.png)

We may need to research options for removing as many of those input fields as we can... Also still seeing the inserted border /cell spacing etc.

![wysiwyg-update-1 1](https://cloud.githubusercontent.com/assets/1927902/11575473/3a315d54-99df-11e5-82d4-9238a16e841f.png)

**Comment by tqureshi-uog** [comment]
I remember working to resolve this issue on the old responsive theme. I don't believe it's possible to remove those default values without hacking ckeditor config files (not a good thing to do since the changes would be overwritten every time ckeditor gets an update). So the workaround I came up with was to configure the Filtered HTML text format to automatically strip all cellspacing, cellpadding, height and width markup from <table>

----- Original Message -----

**Comment by brharp** [comment]
As Tanya mentioned, those height/width/cell spacing/cell padding properties will appear in CKEditor up until the first save. After the first save, the text filter will remove them.

To test:

0. Update issue174 branch.
1. Create a new basic page.
2. Change text format to Filtered HTML.
3. Insert a table and image.
4. Save

Expected result:

- After save, border, cell padding, cell spacing, width and height attributes should be removed.

Note: I have also allowed the class attribute on tables and images, to enable use of the 'table' and 'img-*' classes.

**Comment by mmafe** [comment]
Followed these steps:

1. Checked out branch issue174.
2. Added new basic page.
3. Switched format from Full HTML to Filtered HTML

**Sub-Result:** Image button is available, but table button is missing.

![wysiwyg-update-1 2](https://cloud.githubusercontent.com/assets/1927902/11598104/efb9cb30-9a8d-11e5-8fba-405954daeae1.png)


4. Inserted image.
5. Added img-responsive bootstrap class to image.
6. Added border, height, hspace, vspace
7. Saved.

**Result:** Image appeared. Border, height, hspace, vpsace stripped. 

Img-responsive bootstrap class stripped as well.

![wysiwyg-update-1 3](https://cloud.githubusercontent.com/assets/1927902/11598092/da3945b0-9a8d-11e5-9c6b-17f775cdd819.png)

**Comment by brharp** [comment]
@mmafe did you clear your cache after switching branches? I tried starting from the develop branch and switching to branch issue174. The table button did not appear until I cleared all caches.

**Comment by mmafe** [comment]
#Meeting Notes
From @anncesar, @brharp, and @mmafe

**What's done...**
- Filtered HTML allows for tables and images >> we'll need to re-add the buttons from Full HTML

**What we can do**
- Full HTML is only available for the site manager
- Filtered HTML is available to all other roles
- Allow default dropdown to be "Filtered HTML" for all roles

**Options we'd like to see  if possible**
- Check to see if we can remove the default values
- Image: Remove the width, height, border input fields
- Table: Remove border, alignment, width, height, cellspacing, cell padding
- Image: Add img-responsive class (if possible)
- Table: Add table class (if possible)

**Comment by mancusop** [comment]
If we can really do:
* Image: Add img-responsive class (if possible)
* Table: Add table class (if possible)

And have a patch to apply whenever ckeditor gets updated we may limit a lot of potential future accessibility issues.

What about a javascript checking/injection function or a Drupal function that checks all images and tables when the save button is pressed and adds the classes as required?

**Comment by mmafe** [comment]
Related to issue #16

**Comment by mancusop** [comment]
Don't know if related to this issue - Filtered HTML strips out heading tags. Example page
@ aodademo/filtered-html-headings

if edited with Full HTML headings are shown correctly - else all headings are stripped in Filtered mode.

**Comment by mancusop** [comment]
@brharp @mmafe  Are there any updates to ckeditor coming down the line in relation to img-responsive class and table class?
